packages/wallets/src/features/cfd/screens/DynamicLeverage/__tests__/DynamicLeverageScreen.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { useDebounceValue } from 'usehooks-ts';
+import { useDynamicLeverage } from '@deriv/api-v2';
+import { render, screen } from '@testing-library/react';
+import { useDynamicLeverageModalState } from '../../../components/DynamicLeverageContext';
+import DynamicLeverageScreen from '../DynamicLeverageScreen';
+
+jest.mock('usehooks-ts', () => ({
+    useDebounceValue: jest.fn(() => [false]),
+}));
+
+jest.mock('@deriv/api-v2', () => ({
+    useDynamicLeverage: jest.fn(() => ({
+        data: {
+            cryptocurrencies: {
+                display_name: 'Cryptocurrencies',
+                instruments: ['BTC/USD'],
+                max: 500,
+                min: 25,
+                volume: { data: [] },
+            },
+            forex: {
+                display_name: 'Forex',
+                instruments: ['EUR/USD'],
+                max: 1500,
+                min: 100,
+                volume: { data: [] },
+            },
+            metals: {
+                display_name: 'Metals',
+                instruments: ['Gold'],
+                max: 1000,
+                min: 50,
+                volume: { data: [] },
+            },
+            stock_indices: {
+                display_name: 'Stock Indices',
+                instruments: ['US30'],
+                max: 200,
+                min: 10,
+                volume: { data: [] },
+            },
+        },
+    })),
+}));
+
+jest.mock('../../../components/DynamicLeverageContext', () => ({
+    useDynamicLeverageModalState: jest.fn(() => ({
+        isDynamicLeverageVisible: true,
+    })),
+}));
+
+jest.mock('../DynamicLeverageMarketCard', () => ({
+    DynamicLeverageMarketCard: ({ displayName }: { displayName: string }) => (
+        <div data-testid='dt_dynamic_leverage_market_card'>{displayName}</div>
+    ),
+}));
+
+describe('DynamicLeverageScreen', () => {
+    it('renders content of dynamic leverage screen', () => {
+        render(<DynamicLeverageScreen />);
+
+        expect(screen.getByText(/Enjoy dynamic leverage/)).toBeInTheDocument();
+        expect(screen.getByText('up to 1:1500')).toBeInTheDocument();
+        expect(screen.getAllByTestId('dt_dynamic_leverage_market_card')[0]).toBeInTheDocument();
```

Comment:
Since we're testing for content here @lubega-deriv, wouldn't it be better to check for the total number of cards. i.e. If all the cards are rendered? If there are supposed to be 3 cards, then one is not rendered, the test won't catch it, yet in actual sense, it should right?
```suggestion
        expect(screen.getAllByTestId('dt_dynamic_leverage_market_card')[0]).toBeInTheDocument();
```